synfig-core/src/synfig/rendering/software/surfacesw.h

```diff
@@ -46,28 +46,35 @@ class SurfaceSW: public Surface
 {
 public:
 	typedef etl::handle<SurfaceSW> Handle;
+	static Token token;
+	virtual Token::Handle get_token() const
+		{ return token.handle(); }
 
 private:
 	bool own_surface;
 	synfig::Surface *surface;
 
 protected:
-	virtual bool create_vfunc();
+	virtual bool create_vfunc(int width, int height);
 	virtual bool assign_vfunc(const Surface &surface);
-	virtual void destroy_vfunc();
-	virtual bool get_pixels_vfunc(Color *buffer) const;
+	virtual bool clear_vfunc();
+	virtual bool reset_vfunc();
+	virtual const Color* get_pixels_pointer_vfunc() const;
 
 public:
 	SurfaceSW();
-	explicit SurfaceSW(const Surface &other);
+	explicit SurfaceSW(const Surface &other, bool own_surface);
 	~SurfaceSW();
```

Comment:
Didn't see here surface width and height, and it is unclear where i can get them